packages/amplify_datastore/lib/types/DataStoreHubEvents/ModelSyncedEvent.dart

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'HubEventPayload.dart';
+
+class ModelSyncedEvent extends HubEventPayload {
+  String modelName;
+  bool isFullSync;
+  bool isDeltaSync;
+  int added;
+  int updated;
+  int deleted;
+  ModelSyncedEvent(Map<dynamic, dynamic> serializedData) {
+    modelName = serializedData["model"];
+    isFullSync = serializedData["isFullSync"];
+    isDeltaSync = serializedData["isDeltaSync"];
+    added = serializedData["added"];
+    updated = serializedData["updated"];
+    deleted = serializedData["deleted"];
+  }
```

Comment:
For every new class introduces in this PR, you should include equals, tostring, and hash. Anything that a customer is expected to directly interact with should have all three.

Reply:
This is a good ask - I started implementing, but it's enough work to warrent a different PR.  I've added it to our backlog.